src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const res = [];
+
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
+      Object.assign(stateClone, i.extraData);
+    }
+
+    if (i.type === 'removeProperties') {
+      for (const j of i.keysToRemove) {
+        delete stateClone[j];
+      }
+    }
+
+    if (i.type === 'clear') {
+      Object.keys(stateClone).forEach((key) => delete stateClone[key]);
+    }
+
+    res.push({ ...stateClone });
+  }
+
+  return res;
 }
```

Comment:
[DONT REPEAT YOURSELF] - If you perform same action in all `switch` cases - do it just once afterwards. In this case, you can push `stateCopy` to `res` once after the `switch` statement, at the end of each loop iteration.